### PR TITLE
fix: Searching for the user in LDAP is much faster than dumping all the us…

### PR DESCRIPTION
**Pull request by davidediger**
…ers and iterating server-side.

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: html/includes/authentication/ldap.inc.php [hunk 1]
```diff
@@ -217,10 +217,42 @@ function can_update_users()
...
+                        $userlist[$username] = array(
+                            'username' => $username,
+                            'realname' => $realname,
+                            'user_id' => $user_id,
+                            'email' => $email,
+                        );
+                    }
+                }
+            }
             return $user;
```
Comment (murrant) [inline html/includes/authentication/ldap.inc.php]: $user doesn't seem to be set?  Also, the return should probably be inside the foreach loop.

**Comment by Zmegolaz** [comment]
This is almost identical to the `get_userlist()` function, how about breaking it out to a separate function and reuse that in both? Or put `$filter` as an optional argument to either `get_user()` or `get_userlist()` and call that one from the other.

**Comment by murrant** [comment]
@Zmegolaz It would need to be a third function.  You can't change the signature of get_user() and get_userlist().

File: html/includes/authentication/ldap.inc.php (hunk 1 above)
Comment (davidediger) [reply html/includes/authentication/ldap.inc.php]: Fixed the return in the latest commit.  The LDAP query should only return 1 result, so inside or outside the foreach shouldn't matter.  I'm not sure what we should do if the LDAP query returns more than one result.

**Comment by davidediger** [comment]
get_userlist() and get_user() are substantially similar except for a) the LDAP filter and b) returning one user vs an array of users.  a) is relatively easy to abstract.  b) is going to require additional engineering.  Agree we should avoid code duplication.  I just wanted to push what we're running in production to get a conversation started.

**Comment by murrant** [comment]
@davidediger The authentication refactor has been merged, can you update this to the new classes?

**Comment by laf** [comment]
@davidediger Any update on this?

**Comment by laf** [comment]
No response on this so far. Closing until we hear back from @davidediger

**Pull request closed without merging**